Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1614,7 +1618,12 @@ void RenderLayer::updateDescendantDependentFlags()
         bool hasNotIsolatedBlendingDescendants = false;
 #endif
 
-        for (RenderLayer* child = firstChild(); child; child = child->nextSibling()) {
+        auto firstLayerChild = [&] () -> RenderLayer* {
+            if (renderer().style().contentVisibility() != ContentVisibility::Visible)
```

Comment:
What happens if we call `updateDescendantDependentFlags()` on a layer that's inside, but not the root of a contentVisibility subtree?

Reply:
In that case the calculation proceeds as normal and since all the skipped content layers have m_hasVisibleContent set to false the calculation will result in hasVisibleDescendant remaining false for the root. So in a way it is doing unneeded work, OTOH adding another condition in the firstLayerChild lambda, while fixing the unneeded work, slows down the common case. I am not sure what option is better.